src/libc/mod.rs

```diff
@@ -0,0 +1,394 @@
+            | CLONE_THREAD | CLONE_SYSVSEM | CLONE_SETTLS
+            | CLONE_CHILD_CLEARTID | CLONE_PARENT_SETTID;
+        mmap(ptr::null_mut(), 0, PROT_READ | PROT_WRITE, MAP_PRIVATE |
+             MAP_ANON, -1, 0);
+    }
+    unimplemented!();
+}
+// Heavily simplified (i.e. less features) version of pthread_create.
+// musl: src/thread/pthread_create.c
+//
+// Doesn't care about signals, thread-local storage and perhaps other things
+// yet.
+#[cfg(any(target_arch = "aarch64",
+          target_arch = "arm",
+          target_arch = "powerpc",
+          target_arch = "x86",
+          target_arch = "x86_64"))]
+pub unsafe fn pthread_create(pthread: *mut pthread_t,
+                             attr: *const pthread_attr_t,
+                             start_routine: extern "C" fn(*mut c_void) -> *mut c_void,
+                             arg: *mut c_void)
+    -> c_int
+{
+    let _ = attr;
+    let flags = CLONE_VM | CLONE_FS | CLONE_FILES | CLONE_SIGHAND
+        | CLONE_THREAD | CLONE_SYSVSEM | CLONE_SETTLS
+        | CLONE_CHILD_CLEARTID | CLONE_PARENT_SETTID;
+
+    let align = 16;
+    let mask = align - 1;
+    let stack_size = ((*attr).stack_size + mask) & !mask;
+
+    // TODO(steed, #131): Make sure this is freed on normal and abnormal thread
+    // exit.
+    let map = mmap(ptr::null_mut(),
+                   stack_size + mem::size_of::<thread>(),
+                   PROT_READ | PROT_WRITE,
+                   MAP_PRIVATE | MAP_ANON,
+                   -1,
+                   0);
+
+    // musl: src/internal/__syscall_ret.c
+    if map as usize > -4096isize as usize {
+        return -(map as c_int);
+    }
+
+    let stack = map.offset(stack_size as isize);
+    let thread = stack as *mut thread;
+    (*thread).this = thread;
+
+    let child_tid = syscall_clone(start_routine,
+                                  stack,
+                                  flags,
+                                  arg,
+                                  &mut (*thread).thread_id,
+                                  thread as *mut c_void,
+                                  &mut (*thread).thread_id);
+    if child_tid < 0 {
+        return -child_tid;
+    }
+    *pthread = pthread_t {
+        thread: thread,
+    };
+    0
+}
+
+#[cfg(any(target_arch = "aarch64",
+          target_arch = "arm",
+          target_arch = "powerpc",
+          target_arch = "x86",
+          target_arch = "x86_64"))]
+extern {
+    // Defined in internal/<arch>.rs.
+    //
+    // Does the equivalent of
+    //
+    // ```
+    // let result = CLONE(flags, child_stack, ptid, ctid, newtls);
+    // if result == 0 {
+    //     EXIT(fn_(arg));
+    // } else {
+    //     result
+    // }
+    // ```
+    //
+    // where CLONE and EXIT are the respective syscalls. Because it mangles
+    // with the stack (in two processes at once), it needs to be written in the
+    // target's assembly.
+    #[link_name = "__steed_clone"]
```

Comment:
Yes, that would be possible, but not zero-cost. :) It would actually pass that parameter at runtime in this case.